hbase-server/src/test/java/org/apache/hadoop/hbase/master/assignment/TestRegionSplit.java

```diff
@@ -146,8 +152,70 @@ public void testSplitTableRegion() throws Exception {
     UTIL.getAdmin().enableTable(tableName);
     Thread.sleep(500);
 
-    assertEquals("Table region not correct.", 2,
-        UTIL.getHBaseCluster().getRegions(tableName).size());
+    List<HRegion> tableRegions = UTIL.getHBaseCluster().getRegions(tableName);
+    assertEquals("Table region not correct.", 2, tableRegions.size());
+    Map<RegionInfo, ServerName> regionInfoMap = UTIL.getHBaseCluster().getMaster().getAssignmentManager()
+      .getRegionStates().getRegionAssignments();
+    assertEquals(regionInfoMap.get(tableRegions.get(0).getRegionInfo()),
+      regionInfoMap.get(tableRegions.get(1).getRegionInfo()));
+  }
+
+  @Test
+  public void testSplitTableRegionAndSeparateChildRegions() throws Exception {
+    cleanupTest();
```

Comment:
You should make a new test suite with this test in it rather than do this cleanupTest and setupCluster... These methods are for before the test class and after... not for calling in the test as here.

Reply:
Thanks, I'll change it.